### PR TITLE
feat: Add `og:locale`

### PR DESCRIPTION
**Pull request by Panquesito7**
**What change does this pull request introduce?**

- Add `og:locale`.

**Checklist**

- [x] I worked on a branch other than `main`.
- [x] My branch is up-to-date with the Upstream `main` branch.
- [ ] I have fixed potential errors using `yarn lint`.
- [ ] I ran `yarn build` to check everything still builds successfully.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/website/pull/141"></a>

### CODE REVIEW

File: components/head.tsx [hunk 1]
```diff
@@ -19,6 +19,14 @@ export default function Head({
...
       <meta property="og:url" content="https://the-algorithms.com" />
       <meta property="og:type" content="website" />
+      <meta property="og:locale" content="en_US" />
+      <meta property="og:locale:alternate" content="de_DE" />
+      <meta property="og:locale:alternate" content="es_MX" />
+      <meta property="og:locale:alternate" content="eo" />
+      <meta property="og:locale:alternate" content="uk" />
+      <meta property="og:locale:alternate" content="hr" />
+      <meta property="og:locale:alternate" content="zh_Hans" />
+      <meta property="og:locale:alternate" content="it" />
```
Comment (maltejur) [inline components/head.tsx]: I would suggest generating the locales automatically, this way they don't have to be updated.

```suggestion
      {locales.map((locale) => (
        <meta
          property={
            router.locale === locale ? "og:locale" : "og:locale:alternative"
          }
          content={locale}
          key={locale}
        />
      ))}
```

You would also have to add `import locales from "lib/locales";` at the top of the document and `const router = useRouter();` in the component funtion.

**Comment by Panquesito7** [comment]
Any ideas why it's failing? @maltejur https://vercel.com/thealgorithms/website/6vr6xyUFRYMZZvfivMLEM8s9VG2v

**Comment by maltejur** [comment]
https://github.com/TheAlgorithms/website/pull/141/files#diff-4c640ab8155f9024dcf976f61df427c0685b53662b97a7cb9202c32af4fde158R27

You still need to add the `const router = useRouter();`

**Review by maltejur — APPROVED** [review]
Looks good. Since you already added the OpenGraph tags, maybe a better og preview picture would be nice.

**Pull request merged**